libevmasm/RuleList.h

```diff
@@ -391,6 +391,31 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	rules.push_back({
+		// MUL(X, SHL(1, Y)) -> SHL(X, Y)
```

Comment:
Arguments for shift need to be reversed.

Reply:
Haha, duh, right, yes!